docs/topics/native/native-app-with-c-and-libcurl.md

```diff
@@ -0,0 +1,260 @@
+>
+{type="note"}
+
+## Add interoperability to the build process
+
+To use header files, let's make sure they are generated as a part of the build process. For this, add the following
+entry to the `build.gradle(.kts)` file:
+
+<tabs>
+
+```kotlin
+nativeTarget.apply {
+    compilations.main { // NL
+        cinterops {     // NL
+            libcurl     // NL
+        }               // NL
+    }                   // NL
+    binaries {
+        executable { 
+            entryPoint = "main"
+        }
+    }
+}
+    
+```      
+
+```groovy
+nativeTarget.with {
+    compilations.main { // NL
+        cinterops {     // NL
+            libcurl     // NL
+        }               // NL
+    }                   // NL
+    binaries {
+        executable {
+            entryPoint = 'main'
+        }
+    }
+}
+```
+
+</tabs>
+
+The new lines are marked with `// NL`. First, `cinterops` is added, and then an entry for each `def` file. By default,
+the name of the file is used. You can override this with additional parameters:
+
+```groovy
+libcurl {
+    defFile project.file("libcurl.def")
+    packageName 'com.jetbrains.handson.http'
+    compilerOpts '-I/path'
+    includeDirs.allHeaders("path")
+}
+```
+
+See the [Interoperability with C](native-c-interop.md) section for more details on the available options.
+
+## Write the application code
+
+Now you have the library and the corresponding Kotlin stubs and can consume them from your application. For this tutorial,
+let's convert the [simple.c](https://curl.haxx.se/libcurl/c/simple.html) example to Kotlin.
+
+In the `src/nativeMain/kotlin/` folder, update your `Main.kt` file with the following code:
+
+```kotlin
+import kotlinx.cinterop.*
+import libcurl.*
+
+fun main(args: Array<String>) {
+    val curl = curl_easy_init()
+    if (curl != null) {
+        curl_easy_setopt(curl, CURLOPT_URL, "https://example.com")
+        curl_easy_setopt(curl, CURLOPT_FOLLOWLOCATION, 1L)
+        val res = curl_easy_perform(curl)
+        if (res != CURLE_OK) {
+            println("curl_easy_perform() failed ${curl_easy_strerror(res)?.toKString()}")
+        }
+        curl_easy_cleanup(curl)
+    }
+}
+```
+
+As you can see, explicit variable declarations are eliminated in the Kotlin version, but everything else is pretty much
+the same as the C version. All the calls you'd expect in the `libcurl` library are available in the Kotlin equivalent.
+
+> This is a line-by-line literal translation. You could also write this in a more Kotlin idiomatic way.
+>
+
+## Compile and run the application
+
+1. Compile the application. To do that, run the following command in the terminal:
+
+```bash
+./gradlew runDebugExecutableNative
+```
+In this case, the `cinterop` generated part is implicitly included in the build.
+
+2. If there are no errors during compilation, run the build by creating a [Run Configuration](https://www.jetbrains.com/help/idea/run-debug-configuration.html#create-permanent).
```

Comment:
Maybe some more explanation here, how to run?

Let me explain my idea — I really want readers here to run the app easily, without going to other docs.
And if there is a chance to show here how to create a configuration right for this app, let's provide it.

It's the last step of the tutorial, and I expect that they really want to start it here, without reading any additional info on Run configuration.